File: Android/app/src/main/java/knk/ee/neverland/api/AuthAPI.java
package knk.ee.neverland.api;

import knk.ee.neverland.exceptions.LoginFailedException;
import knk.ee.neverland.exceptions.RegistrationFailedException;
import knk.ee.neverland.pojos.RegistrationData;

public interface AuthAPI {
    String attemptLogin(final String login, final String password) throws LoginFailedException;

    void registerAccount(final RegistrationData registrationData) throws RegistrationFailedException;

    boolean isKeyActive(String key);
}

File: Android/app/src/main/java/knk/ee/neverland/api/AuthAPI.kt
package knk.ee.neverland.api

import knk.ee.neverland.api.models.RegistrationData
import knk.ee.neverland.exceptions.APIException

interface AuthAPI {
    @Throws(APIException::class)
    fun attemptLogin(login: String, password: String): Pair<String, Int>

    @Throws(APIException::class)
    fun registerAccount(registrationData: RegistrationData): Pair<String, Int>

    fun getUserID(token: String): Int
}
